public/docs/ts/latest/guide/router.jade

```diff
@@ -1096,8 +1112,14 @@ code-example(language="bash").
 +makeExcerpt('app/heroes/hero-list.component.ts (import)', 'import-router')
 
 :marked
+  We'll import the `switchMap` operator to perform operations on our `Observable`
+  of route parameters.
+
++makeExcerpt('app/heroes/hero-list.component.ts (rxjs imports)', 'rxjs-imports')
+
+:marked
   Then we use the `ActivatedRoute` to access the `params` _Observable_ so we can subscribe
```

Comment:
This sentence (which I probably wrote originally) needs to be reworked. Too wooden, too complicated.

Try:
--
Then we inject the `ActivatedRoute` in the `HeroListComponent` constructor.

[show the code snippet]

The  `ActivatedRoute.params` property is an `Observable` of route parameters. The `params` emits new `id` values when the user navigates to the component.  In `ngOnInit` we subscribe to those values, set the `selectedId`, and get the heroes. 
--
Also re-title the code snippet which purports to be about the constructor but is actually about both the constructor and the `ngOnInit`.